route/build/src/globalData.f90

```diff
@@ -139,8 +139,25 @@ MODULE globalData
   integer(i4b),                    public :: pio_rearranger    = 2              ! 0=>PIO_rearr_none 1=> PIO_rearr_box 2=> PIO_rearr_subset
   integer(i4b),                    public :: pio_root          = 1              !
   integer(i4b),                    public :: pio_stride        = 1              !
-  type(iosystem_desc_t),           public :: pioSystem                          ! PIO I/O system data
-
+  type(iosystem_desc_t), target,   public :: pioSystem                          ! PIO I/O system data
+  ! pio decomposition used for history file variables
+  type(io_desc_t),                 public :: ioDesc_hru_float
+  type(io_desc_t),                 public :: ioDesc_rch_float
+  type(io_desc_t),                 public :: ioDesc_gauge_float
+   ! decomposition used for restart variables
```

Comment:
Something that would be good to have here is to be able to easily flip the storage type between float and double. This is something we want to do in the future. Restarts should always be double, but history you want to be able to easily choose between float and double. You might still have the explicit double and float variables. But, maybe for example you keep the ioDescRchHist variable, but have it assigned to the double or float specific variable at a higher level at initialization? Then to flip between the types you can do it at initialization and not have to change other parts of the code...

Reply:
This will be worked on more later.